firebase_admin/_sseclient.py

```diff
@@ -0,0 +1,198 @@
+"""SSEClient module to handle streaming of realtime changes on the database
+to the firebase-admin-sdk
+"""
+
+import re
+import time
+import warnings
+import six
+
+import requests
+
+
+# Technically, we should support streams that mix line endings.  This regex,
+# however, assumes that a system will provide consistent line endings.
+end_of_field = re.compile(r'\r\n\r\n|\r\r|\n\n')
+
+
+class KeepAuthSession(requests.Session):
+    """A session that does not drop Authentication on redirects between domains"""
+    def rebuild_auth(self, prepared_request, response):
+        pass
+
+
+class SSEClient(object):
+    """SSE Client Class"""
+    def __init__(self, url, session, build_headers, last_id=None, retry=3000, **kwargs):
+        self.should_connect = True
+        self.url = url
+        self.last_id = last_id
+        self.retry = retry
+        self.running = True
+        # Optional support for passing in a requests.Session()
+        self.session = session
+        # function for building auth header when token expires
+        self.build_headers = build_headers
+        self.start_time = None
+        # Any extra kwargs will be fed into the requests.get call later.
+        self.requests_kwargs = kwargs
+
+        # The SSE spec requires making requests with Cache-Control: nocache
+        if 'headers' not in self.requests_kwargs:
+            self.requests_kwargs['headers'] = {}
+        self.requests_kwargs['headers']['Cache-Control'] = 'no-cache'
+
+        # The 'Accept' header is not required, but explicit > implicit
+        self.requests_kwargs['headers']['Accept'] = 'text/event-stream'
+
+        # Keep data here as it streams in
+        self.buf = u''
+
+        self._connect()
+
+    def close(self):
+        """Close the SSE Client instance"""
+        # TODO: check if AttributeError is needed to catch here
+        self.should_connect = False
+        self.retry = 0
+        self.resp.close()
+        #  self.resp.raw._fp.fp.raw._sock.shutdown(socket.SHUT_RDWR)
```

Comment:
Remove the commented out lines?

On a side note, I was testing this branch the other day, and I noticed that calling close() here does not immediately release the underlying socket. As a result the Stream remains active for a while even after calling `stream.close()`. So perhaps we do need the commented out lines?

Reply:
I also noticed the delay in calling `stream.close()`. I was calling it in a single thread, and it takes a few seconds to close and return. I'll test it out a bit more after I followed your other suggestions first.